executing/_position_node_finder.py

```diff
@@ -703,6 +735,57 @@ def node_match(node_type: Union[Type, Tuple[Type, ...]], **kwargs: Any) -> bool:
             if node_match(ast.FormattedValue) and inst_match("FORMAT_VALUE"):
                 return
 
+        if sys.version_info >= (3, 13):
+            if (
+                node_match(ast.FunctionDef)
+                and inst_match("LOAD_CONST")
+                and node.name == instruction.argval.co_name
+            ):
+                return
+
+            if inst_match("NOP"):
+                return
+
+            if inst_match("TO_BOOL") and node_match(ast.BoolOp):
+                return
+
+            if inst_match("CALL_KW") and node_match((ast.Call, ast.ClassDef)):
+                return
+
+            if inst_match("LOAD_FAST", argval=".type_params"):
+                return
+
+            if inst_match("LOAD_FAST", argval="__classdict__"):
+                return
+
+            if inst_match("LOAD_FAST") and node_match(
+                (
+                    ast.FunctionDef,
+                    ast.ClassDef,
+                    ast.TypeAlias,
+                    ast.TypeVar,
+                    ast.Lambda,
+                    ast.AsyncFunctionDef,
+                )
+            ):
+                # closures
+                # TODO: better check that this is actualy a closure for a scope of a type variable
```

Comment:
Can this TODO be resolved? This condition looks very lenient

Reply:
The following code shows an example:
![image](https://github.com/user-attachments/assets/2394888f-b749-4595-a253-54b2f1bda507)

The code is not only generated for type variables (I fixed the comment).

Checking the variable name could be expensive to compute, because we would have to check large ast trees for variable names.

We could check for the LOAD_FAST,BUILD_TUPLE,LOAD_CONST,MAKE_FUNCTION,SET_FUNCTION_ATTRIBUTE pattern and allow the LOAD_FAST in these cases.

Reply:
Please just leave a more detailed comment, maybe a link to this discussion.